src/util.rs

```diff
@@ -41,13 +41,23 @@
 #[macro_export]
 macro_rules! script_with_data_offset {
     ($offset:ident, $script:expr) => {{
-        use fuel_types::{bytes, Immediate18};
-        use $crate::consts::VM_TX_MEMORY;
-        use $crate::prelude::Transaction;
-        let $offset = 0 as Immediate18;
-        let script_bytes: Vec<u8> = { $script }.into_iter().collect();
-        let data_offset = VM_TX_MEMORY + Transaction::script_offset() + bytes::padded_len(script_bytes.as_slice());
-        let $offset = data_offset as Immediate18;
+        let $offset = {
```

Comment:
Any local vars in the script will be eagerly evaluated before such a function would be called, which means the user would have to go back and edit each op-code in their script that needs a data offset.

By using the macro, we can use an identifier token instead of a variable within the script, which allows this process to be done automatically.